### PR TITLE
nyan: init at 0.2.0

### PR DESCRIPTION
**Pull request by Grubben**
###### Description of changes

This is nyan a a data description language for the FOSS engine openage

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [*] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I am a noob! Please correct me if I made any mistakes

### CODE REVIEW

**Comment by Grubben** [comment]
Oh no, the indent style isn't correct! Should I make another PR or is there any way to fix this one?

**Comment by iblech** [comment]
Great catch! Yes, it is possible to modify this pull request.

In your local checkout, fix the indentation style.

Then run `git commit -a --amend`. This will change your commit; but this pull request is not yet updated.

Finally run `git push --force-with-lease`. This will push the changed commit to GitHub and this pull request will update accordingly. The "force" is required because usually it is mistake to "rewrite history". However in this typofix situation it is perfectly acceptable and in fact best practice.

Do let me know in case there are any problems with this procedure :-)

File: maintainers/maintainer-list.nix
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:5272]: Split this file into its own commit, with message `maintainer-list: add grubben`

https://stackoverflow.com/questions/6217156/break-a-previous-commit-into-multiple-commits

File: pkgs/development/libraries/nyan/default.nix
Comment (AndersonTorres) [inline pkgs/development/libraries/nyan/default.nix:6]:
```suggestion
, stdenv
, fetchFromGitHub
, clang_14
, cmake
, flex
```

File: pkgs/development/libraries/nyan/default.nix
Comment (AndersonTorres) [inline pkgs/development/libraries/nyan/default.nix:30]: Short `description` should be short, one sentence. There is no problem being a bit vague here:

```suggestion
		description = "A data description language";
```

File: pkgs/development/libraries/nyan/default.nix
Comment (AndersonTorres) [inline pkgs/development/libraries/nyan/default.nix:32]: Break this in 80-column (80 characters) limit.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix
Comment (Grubben) [reply maintainers/maintainer-list.nix:5272]: oh I see, I'm going to try

**Comment by Grubben** [comment]
Okay, I'm pretty sure this is ready!

File: maintainers/maintainer-list.nix
Comment (Grubben) [reply maintainers/maintainer-list.nix:5272]: Done!

File: pkgs/development/libraries/nyan/default.nix
Comment (AndersonTorres) [inline pkgs/development/libraries/nyan/default.nix:16]: Risking being redundant, the upstream project is not providing tags? It is suboptimal to use raw Git revision hashes as addresses.

File: pkgs/development/libraries/nyan/default.nix
Comment (AndersonTorres) [inline pkgs/development/libraries/nyan/default.nix:35]:
```suggestion
    license = licenses.lgpl3Plus;
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/nyan/default.nix
Comment (Grubben) [reply pkgs/development/libraries/nyan/default.nix:16]: I tried that and it kept breaking

File: pkgs/development/libraries/nyan/default.nix
Comment (AndersonTorres) [inline pkgs/development/libraries/nyan/default.nix:11]: https://github.com/SFTtech/nyan/tags
The most recent tag is 0.2, not 0.2.0

```suggestion
  version = "0.2";
```

File: pkgs/development/libraries/nyan/default.nix
Comment (AndersonTorres) [inline pkgs/development/libraries/nyan/default.nix:16]:
```suggestion
    rev = "v${version}";
```

Or maybe 

```suggestion
    rev = "refs/tags/${version}"; # because pure ${version} doesn't work
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/nyan/default.nix
Comment (Grubben) [reply pkgs/development/libraries/nyan/default.nix:16]: It won't compile if I do this, so I think we should leave it out

Comment (AndersonTorres) [reply pkgs/development/libraries/nyan/default.nix:16]: This is unusual. Can you show the error log?

Comment (Grubben) [reply pkgs/development/libraries/nyan/default.nix:16]: I don't know why changing this would change this but I think this is beyond me
(Not sure how to show the error log so going to put it directly)

> trying https://github.com/SFTtech/nyan/archive/refs/tags/0.2.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0    14    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download source from any mirror
error: builder for '/nix/store/qjiy2g0l0yiizn2h618pifqbbs7smpab-source.drv' failed with exit code 1;
       last 8 log lines:
       >
       > trying https://github.com/SFTtech/nyan/archive/refs/tags/0.2.tar.gz
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       >   0    14    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 404
       > error: cannot download source from any mirror
       For full logs, run 'nix log /nix/store/qjiy2g0l0yiizn2h618pifqbbs7smpab-source.drv'.
error: 1 dependencies of derivation '/nix/store/vx6zwd0mw76v6pgajyvfrn101345jdjr-nyan-0.2.drv' failed to build

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/nyan/default.nix
Comment (AndersonTorres) [inline pkgs/development/libraries/nyan/default.nix:32]: Break this line, 80 char limit.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by necrophcodr** [comment]
@Grubben is this still relevant, or should we close the PR? It is currently outdated, and would need a rebase before merging, and also a look at the requested changes.

File: pkgs/top-level/all-packages.nix
Comment (necrophcodr) [inline pkgs/top-level/all-packages.nix:37727]: This file may not be entirely in alphabetical order, but there's no need to keep it at way. I'd recommend putting this in a more appropriate position.

**Review by necrophcodr — CHANGES_REQUESTED** [review]

**Comment by Grubben** [comment]
Yes, it was! @necrophcodr. I'll do another PR that's more updated

**Pull request closed without merging**
